control/controlclient/direct.go

```diff
@@ -661,6 +661,7 @@ func (c *Direct) sendMapRequest(ctx context.Context, maxPolls int, cb func(*Netw
 	var lastDERPMap *tailcfg.DERPMap
 	var lastUserProfile = map[tailcfg.UserID]tailcfg.UserProfile{}
 	var lastParsedPacketFilter []filter.Match
+	var collectServices bool
```

Comment:
How does this get the old value if it was unset in this round?

Reply:
This variable is above the loop starting on line 674.

It's assigned within the loop whenever the opt.Bool has a value.

Reply:
Ooh I see